index.d.ts

```diff
@@ -650,13 +655,18 @@ export interface WithT {
   t: TFunction;
 }
 
+/**
+ * Gets the defined translation keys if they have been overridden, 'string' type otherwise
+ */
+type TranslationKeys = I18nTranslationKeys['keys'] extends string ? I18nTranslationKeys['keys'] : string;
```

Comment:
This raises a typescript error, so it should be done another way, but I have no idea how.

(btw, `npm run tsc` should be run in the CI as well, CI checks are green but they shouldn't)

Reply:
(btw, `npm run tsc` should be run in the CI as well, CI checks are green but they shouldn't)

Reply:
I had a long thought about this, and it might have no solution as it is. I can't think of any way to add a strict augmented type that would prevail over the default type.

It works here as it is, but it needs a `@ts-ignore`, and I doubt that it's acceptable.